src/TesApi.Web/BatchScheduler.cs

```diff
@@ -401,7 +409,7 @@ public async Task UploadTaskRunnerIfNeeded(CancellationToken cancellationToken)
         {
             var blobUri = await storageAccessProvider.GetInternalTesBlobUrlAsync(NodeTaskRunnerFilename, cancellationToken);
             var blobProperties = await azureProxy.GetBlobPropertiesAsync(blobUri, cancellationToken);
-            if (!runnerMD5.Equals(blobProperties?.ContentMD5, StringComparison.OrdinalIgnoreCase))
+            if (!(await File.ReadAllTextAsync(Path.Combine(AppContext.BaseDirectory, $"scripts/{NodeTaskRunnerMD5HashFilename}"), cancellationToken)).Trim().Equals(Convert.ToBase64String(blobProperties?.ContentHash ?? []), StringComparison.OrdinalIgnoreCase))
```

Comment:
how is this realted to move away from ADAL?

Reply:
Moving away from ADAL includes moving away from the libraries that depend (directly or indirectly) on ADAL. This is the same functionality in the newer Azure Blob Storage library that works with MSAL (the old one requires ADAL).